### PR TITLE
Cloudrun v2 service

### PR DESCRIPTION
**Pull request by shuyama1**
Internal ticket: b/254103525
API: https://cloud.google.com/run/docs/reference/rest/v2/projects.locations.services

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_cloud_run_v2_service`
```

### CODE REVIEW

File: mmv1/products/cloudrunv2/api.yaml
Comment (shuyama1) [inline mmv1/products/cloudrunv2/api.yaml:93]: `annotation` is currently blocked on b/244872932. Discussed with the service team and decide to omit the field from implementation until the API validation is launched.

File: mmv1/products/cloudrunv2/api.yaml
Comment (shuyama1) [inline mmv1/products/cloudrunv2/api.yaml:235]: `ConflictsWith`, `ExactlyOneOf` and `AtLeastOneOf` within lists are currently not supported at SKD level.

File: mmv1/products/cloudrunv2/api.yaml
Comment (roaks3) [inline mmv1/products/cloudrunv2/api.yaml:39]: Should this be `https://cloud.google.com/run/docs/reference/rest/v2/projects.locations.services` instead?

File: mmv1/products/cloudrunv2/api.yaml
Comment (roaks3) [inline mmv1/products/cloudrunv2/api.yaml:152]: Nit: might want the `# Block on b/244872932 for implementation` comment here too so it doesn't get lost

File: mmv1/products/cloudrunv2/api.yaml
Comment (roaks3) [inline mmv1/products/cloudrunv2/api.yaml:337]: I assume grpc is not supported yet?

File: mmv1/products/cloudrunv2/api.yaml
Comment (roaks3) [inline mmv1/products/cloudrunv2/api.yaml:339]: I'm a little confused why these are needed. Based on your tests, it appears that a probe with an empty object here is acceptable?

File: mmv1/products/cloudrunv2/api.yaml
Comment (roaks3) [inline mmv1/products/cloudrunv2/api.yaml:569]: Should this be `state` (here and under `conditions`)? I'm seeing state instead of status at https://cloud.google.com/run/docs/reference/rest/v2/Condition.

File: mmv1/products/cloudrunv2/api.yaml
Comment (roaks3) [inline mmv1/products/cloudrunv2/api.yaml:669]: It looks like you've used `!ruby/object:Api::Type::Enum` for enum types that are provided by the user, which makes sense to me, but here since it is output only, would a `String` be more appropriate, like you did for the `conditions` object above?

File: mmv1/products/cloudrunv2/terraform.yaml
Comment (roaks3) [inline mmv1/products/cloudrunv2/terraform.yaml:18]: How does this field behave in `terraform.yaml`? It looks the same as in `api.yaml`, so on first glance it seems redundant.

File: mmv1/templates/terraform/examples/cloudrunv2_service_probes.tf.erb
Comment (roaks3) [inline mmv1/templates/terraform/examples/cloudrunv2_service_probes.tf.erb:8]: Nit: looks like a <tab> got in here

**Review by roaks3 — CHANGES_REQUESTED** [review]
Overall this looks great to me. I noticed some small things that might need changes, but the only thing that seems like a potential blocker is a field called `status` which I think needs to be `state`.

File: mmv1/products/cloudrunv2/api.yaml
Comment (shuyama1) [reply mmv1/products/cloudrunv2/api.yaml:153]: Added!

File: mmv1/products/cloudrunv2/api.yaml
Comment (shuyama1) [reply mmv1/products/cloudrunv2/api.yaml:338]: yeah, it doesn't look like it's supported in v2 based on the doc

File: mmv1/products/cloudrunv2/api.yaml
Comment (shuyama1) [reply mmv1/products/cloudrunv2/api.yaml]: Good catch! Fixed!

File: mmv1/products/cloudrunv2/api.yaml
Comment (shuyama1) [reply mmv1/products/cloudrunv2/api.yaml]: fixed

File: mmv1/products/cloudrunv2/terraform.yaml
Comment (shuyama1) [reply mmv1/products/cloudrunv2/terraform.yaml]: Absolutely right! removed!

File: mmv1/products/cloudrunv2/api.yaml
Comment (shuyama1) [reply mmv1/products/cloudrunv2/api.yaml:340]: I think service team confirms that `http_get` should be a valid config.

**Review by roaks3 — APPROVED** [review]
Adding Stephen for follow-up review

File: mmv1/products/cloudrunv2/api.yaml
Comment (melinath) [inline mmv1/products/cloudrunv2/api.yaml:499]:
```suggestion
                              Integer octal mode bits to use on this file, must be a value between 01 and 0777 (octal). If 0 or not set, the Volume's default mode will be used.
```

File: mmv1/products/cloudrunv2/api.yaml
Comment (melinath) [inline mmv1/products/cloudrunv2/api.yaml:495]: I think that HCL may not allow octal integers - possibly we need to represent this with a String?

```suggestion
                          - !ruby/object:Api::Type::String
```

File: mmv1/products/cloudrunv2/api.yaml
Comment (melinath) [inline mmv1/products/cloudrunv2/api.yaml:548]: possibly this should be `default_from_api: true`?

File: mmv1/products/cloudrunv2/api.yaml
Comment (melinath) [inline mmv1/products/cloudrunv2/api.yaml:617]:
```suggestion
              name: "state"
```

File: mmv1/products/cloudrunv2/api.yaml
Comment (melinath) [inline mmv1/products/cloudrunv2/api.yaml:701]: Should the resource wait until not reconciling before updating? similar to go/tpg-resource-state-handling

**Review by melinath — CHANGES_REQUESTED** [review]
a couple small changes, lgtm otherwise.

File: mmv1/products/cloudrunv2/api.yaml
Comment (shuyama1) [reply mmv1/products/cloudrunv2/api.yaml:701]: I double checked and I think this should be handled by waiting on the operation. So maybe we can remove this field?

File: mmv1/products/cloudrunv2/api.yaml
Comment (shuyama1) [reply mmv1/products/cloudrunv2/api.yaml:548]: Good catch! Added!

Comment (shuyama1) [reply mmv1/products/cloudrunv2/api.yaml:548]: Oh wait. Since the default is 0, it may not run into perma-diff though.

File: mmv1/products/cloudrunv2/api.yaml
Comment (shuyama1) [reply mmv1/products/cloudrunv2/api.yaml]: The API returns decimal integer for this field. I'm inclined to keep this field as integer (this is also how this field is implemented in v1) and add an example of this use case ([config](https://github.com/hashicorp/terraform-provider-google/blob/main/google/resource_cloud_run_service_generated_test.go#L432)). Or we can also implement this as string and add a DSF.

Comment (melinath) [reply mmv1/products/cloudrunv2/api.yaml]: If integer works then that's fine with me. 👍

File: mmv1/products/cloudrunv2/api.yaml
Comment (melinath) [reply mmv1/products/cloudrunv2/api.yaml:701]: I'm fine either way

**Review by melinath — APPROVED** [review]
LGTM - it could be nice to have tests for `container.command` and `liveness_probe.tcp_socket` for this resource as well.

**Comment by shuyama1** [comment]
@melinath I suspect the TFV tests failed here maybe due to the TFV build failure for the last two commits.

**Review by melinath — APPROVED** [review]
+1 TFV failure is unrelated - opened https://github.com/GoogleCloudPlatform/magic-modules/pull/6904 to remove that test case for now.

**Comment by shuyama1** [comment]
Only rebase to get TFV fix in. No new code changes and will merge it after tests pass.

**Pull request merged**
